src/action.rs

```diff
@@ -169,3 +174,19 @@ macro_rules! action_impl_future_wrapper {
     };
 }
 pub(crate) use action_impl_future_wrapper;
+
+use crate::Collection;
+
+pub(crate) struct CollRef<'a> {
```

Comment:
The context for this is that there are a bunch of `Collection` methods that don't actually use the document type parameter.  If the action struct for those methods just contained a `&Collection<T>` the action struct would need to have the type parameter as well, which means it inherits a bunch of constraints from `T` that it doesn't actually need and cause various hassle.

The action can just contain a `Collection<Document>` directly, but that means that unlike all the other actions it doesn't have a lifetime parameter binding it to the original call site; the `CollRef` struct here synthesizes that parameter.  I think this is valuable for consistency, and it also means that if the added cost of the `Arc::clone` somehow turns out to be a bottleneck we can potentially change the action to use an (unsafe) ref transmute to avoid the cost.